src/api/app/models/bs_request.rb

```diff
@@ -969,8 +969,12 @@ def auto_accept
     return if approver && state == :review
 
     with_lock do
-      User.current ||= User.find_by_login(creator) if accept_at
-      User.current = User.find_by_login(approver) if approver
+      if accept_at
```

Comment:
not sure it matters, but just for the record: this changes the logic to prefer creator over approver - it preferred approver before.